crates/config/src/rule/range.rs

```diff
@@ -0,0 +1,161 @@
+use ast_grep_core::{meta_var::MetaVarEnv, Doc, Language, Node, Position};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+/// Represents a position in a document
+#[derive(Serialize, Deserialize, Clone, JsonSchema)]
+pub struct SerializablePosition {
+  pub row: usize,
+  pub column: usize,
+}
+
+/// Represents a position in source code using 0-based row and column numbers
+#[derive(Serialize, Deserialize, Clone, JsonSchema)]
+pub struct SerializableRange {
+  /// 0-based row number in the source code
+  pub start: SerializablePosition,
```

Comment:
QQ: will it be useful to skip start or end? Similarly will it be helpful to skip row or column?